framework/include/materials/MaterialBase.h

```diff
@@ -322,10 +318,11 @@ class MaterialBase : public MooseObject,
     PREV
   };
 
-  std::map<std::string, MaterialPropStateInt> _props_to_flags;
+  /// The minimum states requested (0 = current, 1 = old, 2 = older)
+  std::unordered_map<unsigned int, unsigned int> _props_to_min_states;
```

Comment:
Because this is just for error checking and we will have sparse property IDs from within a single material class (this won't own the IDs for _all_ materials that are declared, at least in the case where we have more than one `Material` obejct.

Reply:
That makes sense, but would you add a comment to that effect?  This is the sort of thing I could see myself forgetting and pessimizing years down the road.